sklearn/linear_model/logistic.py

```diff
@@ -425,34 +425,56 @@ def hessp(v):
 
 
 def _check_solver_option(solver, multi_class, penalty, dual):
-    if solver not in ['liblinear', 'newton-cg', 'lbfgs', 'sag', 'saga']:
-        raise ValueError("Logistic Regression supports only liblinear, "
-                         "newton-cg, lbfgs, sag and saga solvers, got %s"
-                         % solver)
 
+    # Default values raises a future warning
+    if solver == 'warn':
+        solver = 'liblinear'
+        warnings.warn("Default solver will be changed to 'lbfgs' in 0.22. "
+                      "Use a specific solver to silence this warning.",
+                      FutureWarning)
+
+    if multi_class == 'warn':
```

Comment:
We should only do this if `len(self.classes_) > 2`

Reply:
Done

Reply:
Are we sure we don't want to also use `multinomial` for binary categorization? In very superficial tests I have run, it appeared to be faster,
```py
In [3]: from sklearn.linear_model import LogisticRegression

In [16]: import numpy as np

In [17]: rng = np.random.RandomState(42)

In [18]: X_train = rng.rand(10000, 100)

In [19]: y_train = rng.randint(2, size=(10000,)) 

In [20]: %timeit LogisticRegression(solver='lbfgs', multi_class='ovr').fit(X_train, y_train)
333 ms ± 1.72 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [21]: %timeit LogisticRegression(solver='lbfgs', multi_class='multinomial').fit(X_train, y_train)
128 ms ± 177 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

for `n_features=10` the difference is even more pronounced,
```py
In [23]: X_train = rng.rand(10000, 10)

In [24]: y_train = rng.randint(2, size=(10000,)) 

In [25]: %timeit LogisticRegression(solver='lbfgs', multi_class='ovr').fit(X_train, y_train)
143 ms ± 174 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [26]: %timeit LogisticRegression(solver='lbfgs', multi_class='multinomial').fit(X_train, y_train)
18.7 ms ± 35.6 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

I have not investigated where this difference is coming from implementation wise.

If we do want to use OVR in the binary, I think it might be worth documenting it (unless it is already).

Reply:
That makes no sense and we should investigate!